test/cases/compile_errors/align_zero.zig

```diff
@@ -1,52 +1,98 @@
-pub var global_var: i32 align(0) = undefined;
+var global_var: i32 align(0) = undefined;
 
-pub export fn a() void {
+export fn a() void {
     _ = &global_var;
 }
 
-pub extern var extern_var: i32 align(0);
+extern var extern_var: i32 align(0);
 
-pub export fn b() void {
+export fn b() void {
     _ = &extern_var;
 }
 
-pub export fn c() align(0) void {}
+export fn c() align(0) void {}
 
-pub export fn d() void {
+export fn d() void {
     _ = *align(0) fn () i32;
 }
 
-pub export fn e() void {
+export fn e() void {
     var local_var: i32 align(0) = undefined;
     _ = &local_var;
 }
 
-pub export fn f() void {
+export fn f() void {
     _ = *align(0) i32;
 }
 
-pub export fn g() void {
+export fn g() void {
     _ = []align(0) i32;
 }
 
-pub export fn h() void {
+export fn h() void {
     _ = struct { field: i32 align(0) };
 }
 
-pub export fn i() void {
+export fn i() void {
     _ = union { field: i32 align(0) };
 }
 
+export fn j() void {
+    _ = @Type(.{ .@"struct" = .{
+        .layout = .auto,
+        .fields = &.{.{
+            .name = "test",
+            .type = u32,
+            .default_value_ptr = null,
+            .is_comptime = false,
+            .alignment = 0,
+        }},
+        .decls = &.{},
+        .is_tuple = false,
+    } });
+}
+
+export fn k() void {
+    _ = @Type(.{ .pointer = .{
+        .size = .one,
+        .is_const = false,
+        .is_volatile = false,
+        .alignment = 0,
+        .address_space = .generic,
+        .child = u32,
+        .is_allowzero = false,
+        .sentinel_ptr = null,
+    } });
+}
+
+export fn l() void {
+    _ = @Type(.{ .@"struct" = .{
+        .layout = .@"packed",
+        .fields = &.{.{
+            .name = "test",
+            .type = u32,
+            .default_value_ptr = null,
+            .is_comptime = false,
+            .alignment = 8,
+        }},
+        .decls = &.{},
+        .is_tuple = false,
+    } });
+}
+
 // error
 // backend=stage2
 // target=native
```

Comment:
Just a note: when you modify test cases, I would appreciate if you could remove these two lines when present. These two values are defaults, and are only written for historical reasons; it'd be better to just get rid of them so these values are only specified in the rare cases where something else is needed. Thanks!